module/zfs/spa.c

```diff
@@ -6115,7 +6133,8 @@ spa_vdev_attach(spa_t *spa, uint64_t guid, nvlist_t *nvroot, int replacing)
 	/*
 	 * Make sure the new device is big enough.
 	 */
-	if (newvd->vdev_asize < vdev_get_min_asize(oldvd))
+	vdev_t *min_vdev = raidz ? oldvd->vdev_child[0] : oldvd;
```

Comment:
This is my ignorance about the internals here, but: is that guaranteed to be the smallest device in a raidz?  It is possible to have, as an example, a raidz with four 1TB drives, and one 8TB drive.

Reply:
Take a look at vdev_get_min_asize() and you'll see that the specific child doesn't actually matter.  Note that this is the same logic for replacing (attaching to) one of the disks in the raidz vdev.  In other words, for size checking purposes, `zpool attach pool raidzN-X new_disk` is the same as `zpool attach pool disk_in_raidz new_disk`.